### PR TITLE
Move components into library directory

### PR DESCRIPTION
**Pull request by brentertz**
### Description

Moves components into `library` directory adjacent to `website` directory

### Motivation and context

We've heard comments from contributors that it is difficult to find some things in our project.  This change is primarily to help make the directory structure a bit easier to understand at first glance.

If we decide at some point to use directory aliases for imports, this will also help in that regard.

Connects #622 

### Screenshots, videos, or demo, if appropriate

https://directories--mineral-ui.netlify.com/

### How to test

1. All build steps work properly
2. No flow errors

### Types of changes

- Other (provide details below)

Change to project directory structure

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support) - **[n/a]**
* [x] Automated tests written and passing - **[n/a]**
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered - **[n/a]**
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered - **[n/a]**
* [x] Documentation created or updated - **[n/a]**
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -19,5 +19,5 @@ suppress_type=$FlowTODO
 module.name_mapper='.*react-docgen-loader.*' -> '<PROJECT_ROOT>/utils/emptyObject'
 module.name_mapper='.*.\(:?md\|svg\)$' -> '<PROJECT_ROOT>/utils/emptyString'
 module.name_mapper='^mineral-ui-icons\/?\(.*\)$' -> '<PROJECT_ROOT>/packages/mineral-ui-icons/src/\1'
-module.name_mapper='^mineral-ui\/?\(.*\)$' -> '<PROJECT_ROOT>/src/\1'
+module.name_mapper='^mineral-ui\/?\(.*\)$' -> '<PROJECT_ROOT>/src/library/\1'
```
Comment (coldpour) [inline .flowconfig]: do we need another path for the website?

Comment (brentertz) [reply .flowconfig]: Not at this time.  We would need it if we want to use shorthand directory aliases for our imports.

Comment (brentertz) [reply .flowconfig]: That being said, would you like shorthand directory aliases for website and library?

Comment (VictoriaVasys) [reply .flowconfig]: I think that would make our files cleaner & more straight-forward. If I remember correctly, @kylegach was thinking about doing this too.

Comment (kylegach) [reply .flowconfig]: I think just the library alias is good for now.

**Pull request merged**